frontend/test/resolution/testInitSemantics.cpp

```diff
@@ -767,15 +767,56 @@ static void testOwnedUserInit(void) {
   std::ignore = resolveModule(ctx, mod->id());
 }
 
-static void testInitFromInit(void) {
-  // test calling 'this.init' from another initializer.
-  Context ctx;
-  Context* context = &ctx;
-  ErrorGuard guard(context);
+// Replaces a placeholder with possible values to help test more programs.
+// If multuple placeholders are present, tests all combinations.
+static std::vector<std::string> getVersionsWithTypes(const std::string& prog,
+                                                     const std::string& placeholder,
+                                                     const std::vector<std::string> types) {
+  std::vector<std::vector<std::string>> variants;
+
+  std::vector<size_t> occurrences;
+  size_t start = 0;
+  while ((start = prog.find(placeholder, start)) != std::string::npos) {
+    occurrences.push_back(start);
+    start += placeholder.length();
+  }
 
-  auto qt = resolveTypeOfXInit(context,
+  std::reverse(occurrences.begin(), occurrences.end());
+  std::vector<std::string> programs = { prog };
+  std::vector<std::string> programsNext;
+
+  for (auto occurrence : occurrences) {
+    for (auto& program : programs) {
+      for (auto& type : types) {
+        std::string newProg = program;
+        newProg.replace(occurrence, placeholder.length(), type);
+        programsNext.push_back(newProg);
+      }
+    }
+    programs = std::move(programsNext);
+  }
+
+  return programs;
+}
+
+static std::vector<std::string> getAllVersions(const std::string& prog) {
+  // Note, const checker currently balks at non-'this' calls to 'init'.
```

Comment:
could you explain the limitation here more? (to me, the comment might be adequate)

Reply:
Actually, I think this is more a problem with the implicit receiver. For some reason, we're emitting an error for `const int` being passed to a `ref myRecord`.

```Chapel
record R {
  var x: int;

  proc init(x: int) {

  }

  proc init(x: int, y: int) {
    init(__primitive("+", x, y));
  }
}
```

Not quite sure the full source of this bug, but it's unrelated and shouldn't be fixed here. I'll open an issue.

Reply:
https://github.com/chapel-lang/chapel/issues/25053